app/src/main/kotlin/com/wire/android/ui/registration/details/CreateAccountDataDetailScreen.kt

```diff
@@ -0,0 +1,416 @@
+    createAccountDataDetailViewModel: CreateAccountDataDetailViewModel = hiltViewModel()
+) {
+    with(createAccountDataDetailViewModel) {
+        fun navigateToCodeScreen() = navigator.navigate(
+            NavigationCommand(
+                CreateAccountVerificationCodeScreenDestination(
+                    createAccountNavArgs.copy(
+                        userRegistrationInfo = createAccountNavArgs.userRegistrationInfo.copy(
+                            email = emailTextState.text.toString().trim(),
+                            name = nameTextState.text.toString().trim(),
+                            password = passwordTextState.text.toString(),
+                            teamName = String.EMPTY
+                        )
+                    )
+                )
+            )
+        )
+
+        LaunchedEffect(createAccountDataDetailViewModel.detailsState.success) {
+            if (createAccountDataDetailViewModel.detailsState.success) navigateToCodeScreen()
+        }
+
+        AccountDetailsContent(
+            state = detailsState,
+            emailTextState = emailTextState,
+            nameTextState = nameTextState,
+            passwordTextState = passwordTextState,
+            confirmPasswordTextState = confirmPasswordTextState,
+            tosUrl = tosUrl(),
+            onTermsDialogDismiss = ::onTermsDialogDismiss,
+            onTermsAccept = ::onTermsAccept,
+            onBackPressed = navigator::navigateBack,
+            onContinuePressed = ::onDetailsContinue,
+            onErrorDismiss = ::onDetailsErrorDismiss,
+            serverConfig = serverConfig
+        )
+    }
+}
+
+@Composable
+private fun AccountDetailsContent(
+    state: CreateAccountDataDetailViewState,
+    emailTextState: TextFieldState,
+    nameTextState: TextFieldState,
+    passwordTextState: TextFieldState,
+    confirmPasswordTextState: TextFieldState,
+    tosUrl: String,
+    onTermsDialogDismiss: () -> Unit,
+    onTermsAccept: () -> Unit,
+    onBackPressed: () -> Unit,
+    onContinuePressed: () -> Unit,
+    onErrorDismiss: () -> Unit,
+    serverConfig: ServerConfig.Links
+) {
+    NewAuthContainer(
+        header = {
+            NewAuthHeader(
+                title = {
+                    Text(
+                        text = stringResource(id = R.string.create_personal_account_title),
+                        style = MaterialTheme.wireTypography.title01
+                    )
+                    if (serverConfig.isOnPremises) {
+                        ServerTitle(
+                            serverLinks = serverConfig,
+                            style = MaterialTheme.wireTypography.body01
+                        )
+                    }
+                },
+                canNavigateBack = true,
+                onNavigateBack = onBackPressed
+            )
+        },
+        contentPadding = dimensions().spacing16x,
+        content = {
+            val keyboardOptions = KeyboardOptions(
```

Comment:
This particular `keyboardOptions` is only used for "name" text field, right? Maybe it would be better to move it there directly because at first I thought we use it for all text fields here 😄